controller_adapters/database.go

```diff
@@ -82,3 +84,102 @@ func (d *Database) GetProfileResolvers() ndb_api.ProfileResolvers {
 	return profileResolvers
 
 }
+
+// Returns basic details about the Time Machine if provided in the
+// underlying database, else returns defaults like:
+// TM Name: db_instance_name_TM
+// TM Description: Time Machine for db_instance_name
+// SLA Name: NONE
+func (d *Database) GetTMDetails() (tmName, tmDescription, slaName string) {
+	tmInfo := d.Spec.Instance.TMInfo
+
+	tmName = tmInfo.Name
+	tmDescription = tmInfo.Description
+	slaName = tmInfo.SLAName
+
+	if tmName == "" {
+		tmName = d.GetDBInstanceName() + "_TM"
+	}
+	if tmDescription == "" {
+		tmDescription = "Time Machine for " + d.GetDBInstanceName()
+	}
+	if slaName == "" {
+		slaName = common.SLA_NAME_NONE
+	}
+
+	return
+}
+
+// Returns a schedule struct for the time machine.
+func (d *Database) GetTMSchedule() (schedule ndb_api.Schedule, err error) {
+	tmInfo := d.Spec.Instance.TMInfo
+	hhmmssDaily := strings.Split(tmInfo.DailySnapshotTime, ":")
+	if len(hhmmssDaily) != 3 {
+		err = errors.New("invalid DailySnapshotTime, use the (24-hour) hh:mm:ss format")
+		return
+	}
+	hh, err := strconv.Atoi(hhmmssDaily[0])
+	if err != nil {
+		err = errors.Join(err, errors.New("error converting daily snapshot time (hh)"))
+		return
+	}
+	mm, err := strconv.Atoi(hhmmssDaily[1])
+	if err != nil {
+		err = errors.Join(err, errors.New("error converting daily snapshot time (mm)"))
+		return
+	}
+	ss, err := strconv.Atoi(hhmmssDaily[2])
+	if err != nil {
+		err = errors.Join(err, errors.New("error converting daily snapshot time (ss)"))
+		return
+	}
+
+	var quarterlySnapshotStartMonth string
+	switch tmInfo.QuarterlySnapshots {
+	case "Jan":
+		quarterlySnapshotStartMonth = "JANUARY"
+	case "Feb":
+		quarterlySnapshotStartMonth = "FEBRUARY"
+	case "Mar":
+		quarterlySnapshotStartMonth = "MARCH"
+	default:
+		quarterlySnapshotStartMonth = ""
+	}
```

Comment:
Open Question: Should we error out if no value is provided?

Reply:
this is an optional input in the spec, right? what happens in practice if the user does not provide this?

Reply:
AFAIK, it gets a default value on the UI.

Reply:
Also, if not provided in yaml input spec it gets defaulted to `Jan`.

Reply:
Update: It only gets defaulted to `Jan` when the entire TMInfo object is not provided in the input spec. otherwise this field remains empty. Hence I feel we should default it to "Jan" explicitly here.